### ISSUE TITLE
Hoppers delete resources when used for a ritual

### ISSUE THREAD

**Issue by Red1004** [Mithion/Mana-And-Artifice#1027]
**Describe the bug**
When using a hopper to insert ingredients into a ritual rune, it will continue to insert the item once the ritual activates (when it raises the items).

**To Reproduce**
1. Prepare a ritual and a hopper to insert resources
2. Insert an excess amount of resources into hopper
3. Proceed with ritual as normal
4. Look at the hoppers inventory and you should see it continually consume items

**Expected behavior**
I expected the hopper to wait until the ritual completes to insert the next item instead of partway through.

**Version Information (please complete the following information):**
 - Mod Version: [3.1.0.1]
 - Forge Version [47.3.0]

**Comment by ShelLuser** [comment]
So, first... your issue is _very_ confusing. Your title suggests that the hopper is removing items, while your post talks about adding an excess of items. And what do you mean with "_raises the items_"?

Anywhoo, what made you conclude that this could be a bug?  Because the way I see it this is normal behavior for a hopper; when an item is stackable then a hopper will max out that stack. 

_However_... I obviously tried to reproduce this (I use the same Forge & MNA versions) but I couldn't:

![2024-12-15_03 30 56](https://github.com/user-attachments/assets/19536036-3be1-4e5b-a861-af915d82e0bf)

Notice my Jade dialog on top?  Chimerite can stack up to 64 yet only 1 item is placed on top of the metal rune. So I can only assume that another mod on your end is interfering with this mechanic, because it most certainly isn't default MNA behavior, as seen above.

I experimented a bit more with this mechanic and noticed that droppers can also be used to place items on top of metal runes. _Disclaimer:_ I play with the Botania mod as well and it changes some of the dropper mechanics so that you can automate planting saplings, I don't know if that had any effect here.

But if hoppers don't work for you, then maybe you could try a dropper?

Hope this can still help!

**Comment by Red1004** [comment]
The ritual needs to be activated in order for the bug to occur. When activated, the items will rise, and at that point the hoppers "see" the rune as empty, and try to fill it, but for whatever reason, the item gets deleted instead.

**Comment by ShelLuser** [comment]
Thanks for the extra explanation Red, very much appreciated!

I can now confirm the issue as well, it's fully reproducable. Just like you said: as soon as the ritual starts then the runes basically become "voids" that gobble up all the available items:

![2024-12-16_03 31 10](https://github.com/user-attachments/assets/63f30254-c4fc-484a-83fe-59e0bc17e54b)

Now, this is just my 2 cents as a fellow player (!) but I'm not sure if this should be considered a bug. Not my call to make of course, but here's the thing: we're playing with magic here, and I could imagine that those mechanics simply don't play by the same rules as those of "technical Minecraft" aka redstone.

I did come up with a possible solution by the way, hope this can be useful:

![2024-12-16_03 44 07](https://github.com/user-attachments/assets/959b051a-1a40-41ce-8cf3-daf1c960540c)

So basically, lock the hoppers and only release them when there's no ritual taking place. If you add a redstone timer you may still be able to automate this whole thing, but that's an assumption on my part.

Funniest thing... Aya & me were busy collecting Ender motes while I checked up on Github for a sec, so I had the perfect opportunity to give this a renewed try as well.  Well, Aya is happy with her free motes 👍

**Comment by Mithion** [comment]
This was resolved in 3.1.0.2
